kafka/consume.go

```diff
@@ -73,29 +75,83 @@ func NewConsumer(log logrus.FieldLogger, conf Config, opts ...ConfigOpt) (*Confl
 		return nil, err
 	}
 
-	confluenceConsumer := &ConfluentConsumer{
-		c:    consumer,
-		conf: conf,
-		log:  log,
+	if conf.Timeout == 0 {
+		conf.Timeout = DefaultTimout
 	}
 
-	return confluenceConsumer, nil
+	part := int32(-1)
+	if conf.Consumer.Partition != nil {
+		part = *conf.Consumer.Partition
+	}
+
+	cc := &ConfluentConsumer{
+		c:         consumer,
+		conf:      conf,
+		log:       log,
+		partition: part,
+	}
+
+	if cc.partition < 0 {
+		// if we are not reading from a specific partition, we subscribe
+		cc.configOnce.Do(func() {
+			cc.log.WithField("kafka_topic", cc.conf.Topic).Debug("Subscribing to Kafka topic")
+			cc.rebalanceHandlerMutex.Lock()
+			defer cc.rebalanceHandlerMutex.Unlock()
+			if err := cc.c.Subscribe(cc.conf.Topic, cc.rebalanceHandler); err != nil {
+				err = errors.Wrap(err, "error subscribing to topic")
```

Comment:
This shadows the `err` I think

Reply:
You mean the scope of `err` is just the If?

Reply:
yup!